Numpy 1.24 compatibility

```
Run pip install --disable-pip-version-check --upgrade pip setuptools==63.* wheel scikit-build 
Requirement already satisfied: pip in /opt/hostedtoolcache/Python/3.[10](https://github.com/nens/threedigrid-builder/actions/runs/3731586042/jobs/6329984381#step:6:11).9/x64/lib/python3.10/site-packages (22.3.1)
Collecting setuptools==63.*
  Using cached setuptools-63.4.3-py3-none-any.whl (1.2 MB)
Collecting wheel
  Using cached wheel-0.38.4-py3-none-any.whl (36 kB)
Collecting scikit-build
  Downloading scikit_build-0.16.4-py3-none-any.whl (78 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.6/78.6 kB 2.3 MB/s eta 0:00:00
Collecting distro
  Using cached distro-1.8.0-py3-none-any.whl (20 kB)
Collecting packaging
  Using cached packaging-22.0-py3-none-any.whl (42 kB)
Installing collected packages: wheel, setuptools, packaging, distro, scikit-build
  Attempting uninstall: setuptools
    Found existing installation: setuptools 65.5.0
    Uninstalling setuptools-65.5.0:
      Successfully uninstalled setuptools-65.5.0
Successfully installed distro-1.8.0 packaging-22.0 scikit-build-0.16.4 setuptools-63.4.3 wheel-0.38.4
Collecting numpy
  Downloading numpy-1.24.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 17.3/17.3 MB 41.4 MB/s eta 0:00:00
Installing collected packages: numpy
Successfully installed numpy-1.24.0
Collecting threedi-modelchecker
  Cloning https://github.com/nens/threedi-modelchecker (to revision master) to /tmp/pip-install-e9jigcfz/threedi-modelchecker_3f[11](https://github.com/nens/threedigrid-builder/actions/runs/3731586042/jobs/6329984381#step:6:12)1bac529247d9baf261534882e524
  Running command git clone --filter=blob:none --quiet https://github.com/nens/threedi-modelchecker /tmp/pip-install-e9jigcfz/threedi-modelchecker_3f111bac529247d9baf261534882e524
  Resolved https://github.com/nens/threedi-modelchecker to commit c2584b472705a09a49f[12](https://github.com/nens/threedigrid-builder/actions/runs/3731586042/jobs/6329984381#step:6:13)da2f28ba2c5336c5a19
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Obtaining file:///home/runner/work/threedigrid-builder/threedigrid-builder
  Checking if build backend supports build_editable: started
  Checking if build backend supports build_editable: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting pygdal==3.4.1.*
  Using cached pygdal-3.4.1.10-cp310-cp310-linux_x86_64.whl
Requirement already satisfied: numpy>=1.0.0 in /opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages (from pygdal==3.4.1.*) (1.24.0)
Collecting Click
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting GeoAlchemy2!=0.11.*,>=0.9
    In: :_fgrid:/home/runner/work/threedigrid-builder/threedigrid-builder/libthreedigrid/_fgrid.pyf:_fgrid:unknown_interface:m_cells:set_2d_computational_nodes_lines
    get_useparameters: no module iso_fortran_env info used by set_2d_computational_nodes_lines
    Applying post-processing hooks...
      character_backward_compatibility_hook
    gmake[2]: *** [CMakeFiles/_fgrid.cpython-310-x86_64-linux-gnu.so.dir/build.make:79: _fgridmodule.c] Error 1
    gmake[1]: *** [CMakeFiles/Makefile2:[13](https://github.com/nens/threedigrid-builder/actions/runs/3731586042/jobs/6329984381#step:6:14)8: CMakeFiles/_fgrid.cpython-310-x86_64-linux-gnu.so.dir/all] Error 2
    gmake: *** [Makefile:136: all] Error 2
    Traceback (most recent call last):
      File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/skbuild/setuptools_wrap.py", line 640, in setup
        cmkr.make(make_args, install_target=cmake_install_target, env=env)
      File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/skbuild/cmaker.py", line 684, in make
        self.make_impl(clargs=clargs, config=config, source_dir=source_dir, install_target=install_target, env=env)
      File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/site-packages/skbuild/cmaker.py", line 7[15](https://github.com/nens/threedigrid-builder/actions/runs/3731586042/jobs/6329984381#step:6:16), in make_impl
        raise SKBuildError(
    
    An error occurred while building with CMake.
      Command:
        cmake --build . --target install --config Release --
      Install target:
        install
      Source directory:
        /home/runner/work/threedigrid-builder/threedigrid-builder
      Working directory:
        /home/runner/work/threedigrid-builder/threedigrid-builder/_skbuild/linux-x86_64-3.10/cmake-build
    Please check the install target is valid and see CMake's output for more information.
    [end of output]

note: This error originates from a subprocess, and is likely not a problem with pip.
Error: Process completed with exit code 1.
```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -4,6 +4,8 @@ Changelog of threedigrid-builder
 1.6.2 (unreleased)
 ------------------
 
+- Fix build with numpy >= 1.24
+
 - Require schema version 214 (threedi-modelchecker >= 0.35).
 
 - Set 1D-2D line dpumax based on v2_potential_breach.exchange_level >

--- a/libthreedigrid/_fgrid.pyf
+++ b/libthreedigrid/_fgrid.pyf
@@ -6,7 +6,6 @@ python module _fgrid ! in
         module m_quadtree ! in :_fgrid:quadtree.f90
             subroutine make_quadtree(kmax,mmax,nmax,lgrmin,use_2d_flow,area_mask,lg,quad_idx,n_cells,n_line_u,n_line_v) ! in :_fgrid:quadtree.f90:m_quadtree
                 use parameters, only : NODATA
-                use iso_fortran_env, only : int16
                 integer intent(in) :: kmax
                 integer dimension(:),intent(in) :: mmax
                 integer dimension(:),intent(in) :: nmax
@@ -21,9 +20,7 @@ python module _fgrid ! in
             end subroutine make_quadtree
         end module m_quadtree
         module m_cells ! in :_fgrid:cells.f90
-            use iso_fortran_env, only: int16
             subroutine set_2d_computational_nodes_lines(origin,lgrmin,kmax,mmax,nmax,dx,lg,nodk,nodm,nodn,quad_idx,bounds,coords,pixel_coords,area_mask,line,cross_pix_coords,n_line_u,n_line_v) ! in :_fgrid:cells.f90:m_cells
-                use iso_fortran_env, only: int16
                 use m_grid_utils, only: get_lg_corners,get_cell_bbox,get_pix_corners,pad_area_mask
                 double precision dimension(2),intent(in) :: origin
                 integer intent(in) :: lgrmin

--- a/libthreedigrid/array_utils.f90
+++ b/libthreedigrid/array_utils.f90
@@ -12,19 +12,19 @@ module m_array_utils
     
     contains
 
-    function check_bounds_d_1d(array, i0, i1) result(check)
+    function check_bounds_d_1d(array, i0, i1) result(inbound)
 
         double precision, intent(in) :: array(:)
         integer, intent(in) :: i0
         integer, intent(in) :: i1
-        logical :: check
+        logical :: inbound
 
-        check = .FALSE.
+        inbound = .FALSE.
         if (i0 >= lbound(array, 1) .and. i1 <= ubound(array, 1)) then
             if(i0 <= i1) then
-                check = .TRUE.
+                inbound = .TRUE.
             else
-                check = .FALSE.
+                inbound = .FALSE.
             endif
         else
             write(*,*) '** WARNING: Array out of bounds'
@@ -32,14 +32,14 @@ module m_array_utils
 
     end function check_bounds_d_1d
 
-    function check_bounds_d_2d(array, i0, i1, axis_in) result(check)
+    function check_bounds_d_2d(array, i0, i1, axis_in) result(inbound)
 
         double precision, intent(in) :: array(:,:)
         integer, intent(in) :: i0
         integer, intent(in) :: i1
         integer, intent(in), optional :: axis_in
         integer :: axis
-        logical :: check
+        logical :: inbound
 
         if (present(axis_in)) then
             axis = axis_in
@@ -47,13 +47,13 @@ module m_array_utils
             axis = 1
         endif
 
-        check = .FALSE.
+        inbound = .FALSE.
         if (i0 >= lbound(array, axis) .and. i1 <= ubound(array, axis)) then
             if(i0 <= i1) then
-                check = .TRUE.
+                inbound = .TRUE.
             else
                 write(*,*) '** WARNING: Indices in incorrect order.'
-                check = .FALSE.
+                inbound = .FALSE.
             endif
         else
             write(*,*) '** WARNING: Array out of bounds'
@@ -61,19 +61,19 @@ module m_array_utils
 
     end function check_bounds_d_2d
 
-    function check_bounds_i_1d(array, i0, i1) result(check)
+    function check_bounds_i_1d(array, i0, i1) result(inbound)
 
         integer, intent(in) :: array(:)
         integer, intent(in) :: i0
         integer, intent(in) :: i1
-        logical :: check
+        logical :: inbound
 
-        check = .FALSE.
+        inbound = .FALSE.
         if (i0>=lbound(array, 1) .and. i1<=ubound(array, 1)) then
             if(i0 <= i1) then
-                check = .TRUE.
+                inbound = .TRUE.
             else
-                check = .FALSE.
+                inbound = .FALSE.
             endif
         else
             write(*,*) '** WARNING: Array out of bounds'
@@ -81,14 +81,14 @@ module m_array_utils
 
     end function check_bounds_i_1d
 
-    function check_bounds_i_2d(array, i0, i1, axis_in) result(check)
+    function check_bounds_i_2d(array, i0, i1, axis_in) result(inbound)
 
         integer, intent(in) :: array(:,:)
         integer, intent(in) :: i0
         integer, intent(in) :: i1
         integer, intent(in), optional :: axis_in
         integer :: axis
-        logical :: check
+        logical :: inbound
 
         if (present(axis_in)) then
             axis = axis_in
@@ -96,12 +96,12 @@ module m_array_utils
             axis = 1
         endif
 
-        check = .FALSE.
+        inbound = .FALSE.
         if (i0>=lbound(array, axis) .and. i1<=ubound(array, axis)) then
             if(i0 <= i1) then
-                check = .TRUE.
+                inbound = .TRUE.
             else
-                check = .FALSE.
+                inbound = .FALSE.
             endif
         else
             write(*,*) '** WARNING: Array out of bounds'

--- a/libthreedigrid/cells.f90
+++ b/libthreedigrid/cells.f90
@@ -1,6 +1,6 @@
 module m_cells
 
-    use iso_fortran_env, only: int16
+    implicit none
 
     contains
 
@@ -8,7 +8,6 @@ module m_cells
         lg, nodk, nodm, nodn, quad_idx, bounds, coords, pixel_coords,&
         area_mask, line, cross_pix_coords, n_line_u, n_line_v)
         !!! Entry point for setting nodes and lines and there necessary attributes.
-        use iso_fortran_env, only: int16
         use m_grid_utils, only : get_lg_corners, get_cell_bbox, get_pix_corners, pad_area_mask
 
         double precision, intent(in) :: origin(2) ! Origin of Quadtree grid
@@ -86,7 +85,6 @@ module m_cells
 
     subroutine set_2d_computational_lines(l_u, l_v, k, m, n, mn, lg, lgrmin, area_mask, quad_idx, nod, line, cross_pix_coords)
 
-        use iso_fortran_env, only: int16
         use m_grid_utils, only : get_lg_corners, get_pix_corners, crop_pix_coords_to_raster
 
         integer, intent(inout) :: l_u

--- a/libthreedigrid/geo_utils.f90
+++ b/libthreedigrid/geo_utils.f90
@@ -1,7 +1,5 @@
 module m_grid_utils
 
-    use iso_fortran_env, only : int16
-
     implicit none
 
     contains
@@ -80,8 +78,6 @@ module m_grid_utils
     end subroutine crop_pix_coords_to_raster
 
     function pad_area_mask(raster, i0, i1, j0, j1) result(padded_raster)
-        
-        use iso_fortran_env, only: int16
 
         integer*2, intent(in) :: raster(:,:)
         integer, intent(in) :: i0
@@ -90,10 +86,6 @@ module m_grid_utils
         integer, intent(in) :: j1
         integer*2, allocatable :: padded_raster(:, :)
         integer :: i_size, j_size, size_raster_i, size_raster_j
-        integer :: i0r, i1r
-        integer :: j0r, j1r
-        integer :: i0p, i1p
-        integer :: j0p, j1p
 
         size_raster_i = size(raster, 1)
         size_raster_j = size(raster, 2)
@@ -199,7 +191,7 @@ module m_grid_utils
         double precision, intent(out), optional :: xcross, ycross
         double precision :: aa0,aa1,bb0,bb1,cc0,cc1,ddi,ds0,ds1,ds2
         
-        !> @details First it finds the corresponding connection.
+        !> details First it finds the corresponding connection.
         !! Second, the orientation of levee with respect to grid is determined
         !! Third, It is determined which velocity points are connected and which flooding treshold it belongs.
         !> @param[in] x0, y0 coordinates of beginning of levee element
@@ -378,10 +370,10 @@ module m_grid_utils
 
     end function geom_in_polygon
 
-	subroutine point_in_polygon(p_x, p_y, poly_x, poly_y, in_polygon)
-	
-	    implicit none
-    
+    subroutine point_in_polygon(p_x, p_y, poly_x, poly_y, in_polygon)
+
+        implicit none
+
         double precision, intent(in) :: p_x
         double precision, intent(in) :: p_y
         double precision, intent(in) :: poly_x(:)
@@ -398,35 +390,35 @@ module m_grid_utils
         in_polygon = -1
         n_vertices = size(poly_x)
 	
-	    do i= 1, n_vertices
-	        xi = poly_x(i) - p_x
-	        yi = poly_y(i) - p_y
+        do i= 1, n_vertices
+            xi = poly_x(i) - p_x
+            yi = poly_y(i) - p_y
 	!CHECK WHETHER THE POINT IN QUESTION IS AT THIS VERTEX.
-	        if( xi==0.0d0.and.yi==0.0d0) then
-	            in_polygon = 0
-	            return
+            if( xi==0.0d0.and.yi==0.0d0) then
+                in_polygon = 0
+                return
             endif
 	!J IS NEXT VERTEX NUMBER OF POLYGON.
-	        j = 1 + mod(i,n_vertices)
-	        xj = poly_x(j) - p_x
-	        yj = poly_y(j) - p_y
+            j = 1 + mod(i,n_vertices)
+            xj = poly_x(j) - p_x
+            yj = poly_y(j) - p_y
 	!IS THIS LINE OF 0 LENGTH ?
             if(xi==xj.and.yi==yj ) then
                 cycle
             endif
-	        ix = xi>=0.0d0
-	        iy = yi>=0.0d0
-	        jx = xj>=0.0d0
-	        jy = yj>=0.0d0
+            ix = xi>=0.0d0
+            iy = yi>=0.0d0
+            jx = xj>=0.0d0
+            jy = yj>=0.0d0
 	!CHECK WHETHER (PX,PY) IS ON VERTICAL SIDE OF POLYGON.
-	        if(xi==0.0d0.and.xj==0.0d0.and.eor(iy,jy)) then
-	            in_polygon = 0
-	            return
+            if(xi==0.0d0.and.xj==0.0d0.and.eor(iy,jy)) then
+                in_polygon = 0
+                return
             endif
 	!CHECK WHETHER (PX,PY) IS ON HORIZONTAL SIDE OF POLYGON.
-	        if(yi==0.0d0.and.yj==0.0d0.and.eor(ix,jx)) then
-	            in_polygon = 0
-	            return
+            if(yi==0.0d0.and.yj==0.0d0.and.eor(ix,jx)) then
+                in_polygon = 0
+                return
             endif
 	!CHECK WHETHER BOTH ENDS OF THIS SIDE ARE COMPLETELY 1) TO RIGHT
 	!OF, 2) TO LEFT OF, OR 3) BELOW (PX,PY).
@@ -434,17 +426,17 @@ module m_grid_utils
                 cycle
             endif
 	!DOES THIS SIDE OBVIOUSLY CROSS LINE RISING VERTICALLY FROM (PX,PY)
-	        if(.not.(iy.and.jy.and.eor(ix,jx))) then
-	            if((yi*xj-xi*yj)/(xj-xi)<0.0d0) then
+            if(.not.(iy.and.jy.and.eor(ix,jx))) then
+                if((yi*xj-xi*yj)/(xj-xi)<0.0d0) then
                     cycle
                 elseif((yi*xj-xi*yj)/(xj-xi)==0.0d0) then
-	                in_polygon = 0
-	                return
-	            else
-	               in_polygon = -in_polygon
+                    in_polygon = 0
+                    return
+                else
+                    in_polygon = -in_polygon
                 endif
-	        else
-	            in_polygon = -in_polygon
+            else
+                in_polygon = -in_polygon
             endif
 	
         enddo

--- a/libthreedigrid/quadtree.f90
+++ b/libthreedigrid/quadtree.f90
@@ -1,17 +1,15 @@
 module m_quadtree
 
     use parameters, only : NODATA
-    use iso_fortran_env, only : int16
 
     implicit none
 
     contains
 
     subroutine make_quadtree(kmax, mmax, nmax, lgrmin, use_2d_flow, area_mask, lg, quad_idx,&
-        n_cells, n_line_u, n_line_v) bind(c, name="make_quadtree")
+        n_cells, n_line_u, n_line_v)
     !!! Entry point for creating quadtree by setting lg array and then finding number of active cells and lines.
         use parameters, only : NODATA
-        use iso_fortran_env, only : int16
 
         integer, intent(in) :: kmax ! Maximum refinement levels
         integer, intent(in) :: mmax(:) ! X Dimension of each refinement level
@@ -45,7 +43,6 @@ module m_quadtree
     !!! Recursive subroutine to set correct refinement levels on lg refinement array.
         use m_grid_utils, only : get_lg_corners
         use parameters, only : NODATA
-        use iso_fortran_env, only : int16
 
         integer, intent(in) :: k
         integer, intent(in) :: m
@@ -117,7 +114,6 @@ module m_quadtree
         use m_grid_utils, only : get_lg_corners, get_pix_corners, crop_pix_coords_to_raster, pad_area_mask
         use m_cells, only : set_2d_computational_lines
         use parameters, only : NODATA
-        use iso_fortran_env, only : int16
 
         integer, intent(in) :: kmax
         integer, intent(in) :: mmax(:)

--- a/setup.py
+++ b/setup.py
@@ -51,7 +51,7 @@ def get_version():
 
 
 install_requires = [
-    "numpy>=1.15,<1.24",
+    "numpy>=1.15",
     "threedi-modelchecker>=0.35",
     "pygeos>=0.11.1",
     "pyproj>=3",

